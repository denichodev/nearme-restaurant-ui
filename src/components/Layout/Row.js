import React from 'react';
import { bool, string, node } from 'prop-types';
import cn from 'classnames';

const Row = props => {
  const extraClass = props.className && props.className;
  const reverseClass = props.reverse && 'reverse';

  return (
    <div className={cn('row', reverseClass, extraClass)}>{props.children}</div>
  );
};

Row.propTypes = {
  className: string,
  reverse: bool,
  children: node,
};

export default Row;
