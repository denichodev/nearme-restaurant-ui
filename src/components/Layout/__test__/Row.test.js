import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Row from '../Row';

describe('<Row />', () => {
  test('renders correctly', () => {
    const wrapper = shallow(
      <Row>
        <div>Children</div>
      </Row>,
    );

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  test('has reverse class', () => {
    const wrapper = shallow(
      <Row reverse>
        <div>Children</div>
      </Row>,
    );

    expect(wrapper.find('.reverse')).toHaveLength(1);
  });

  test('has extra class', () => {
    const wrapper = shallow(
      <Row reverse className="hello-world">
        <div>Children</div>
      </Row>,
    );

    expect(wrapper.find('.hello-world')).toHaveLength(1);
  });
});
