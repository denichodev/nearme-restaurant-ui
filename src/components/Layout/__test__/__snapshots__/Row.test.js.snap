// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row /> renders correctly 1`] = `
<div
  className="row"
>
  <div>
    Children
  </div>
</div>
`;
